engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -530,6 +535,25 @@ private FacetMetadata findFacetMetadata(FacetingRequest facetingRequest, Iterabl
 		return facetMetadata;
 	}
 
+	protected List<IndexManager> getIndexManagers(EntityIndexBinding binding) {
+		FullTextFilterImplementor[] fullTextFilters = getFullTextFilterImplementorsAsArray();
+		List<IndexManager> indexManagers = Arrays.asList( binding.getSelectionStrategy().getIndexManagersForQuery( fullTextFilters ) );
+		return indexManagers;
+	}
+
+	private FullTextFilterImplementor[] getFullTextFilterImplementorsAsArray() {
```

Comment:
FullTextFilter is much too vague, we could be returning half a dozen different things, and none of them would be handled the same way:

 * `Query`
 * `Filter`
 * `FullTextFilterImplementor`
 * `FullTextFilterDef` (!)
 * `FullTextFilter` (!)
 * etc.

I'll remove the "asArray" suffix, though.